pandas/core/internals/blocks.py

```diff
@@ -2248,14 +2249,17 @@ def _astype(self, dtype, **kwargs):
     def _can_hold_element(self, element):
         tipo = maybe_infer_dtype_type(element)
         if tipo is not None:
-            return tipo == _NS_DTYPE or tipo == np.int64
+            return is_dtype_equal(tipo, self.dtype)
+        elif element is NaT:
+            return True
         elif isinstance(element, datetime):
+            if self.is_datetimetz:
+                return tz_compare(element.tzinfo, self.dtype.tz)
             return element.tzinfo is None
         elif is_integer(element):
             return element == tslibs.iNaT
 
-        # TODO: shouldnt we exclude timedelta64("NaT")?  See GH#27297
-        return isna(element)
+        return isna(element) and not isinstance(element, np.timedelta64)
```

Comment:
I think we should make a named function for these, maybe

isna_datetimelike
isna_timedeltalike

(and maybe combine the is_null_datetimelike that's in cython) / I would rename this one actually to isna_datetimelike

rather than repeating these isintance checks

Reply:
yah, i think now that the #27331 is merged we can use `is_valid_nat_for_dtype`